CONTRIBUTING.md

```diff
@@ -0,0 +1,31 @@
+# Contributing to Rust IPFS
+
+Welcome, and thank you for your interest in contributing to Rust IPFS. Issues and pull requests are encouraged.
+
+## Target Build
+
+Rust IPFS will always target the current _stable_ version of Rust that is released. Our CI/CD tests will reflect this.
+
+TODO: Instructions on installing the up-to-date stable toolchain
+TODO: First Principles
+
+## Roadmap
+
+- Link to roadmap issue or project?
```

Comment:
Here I am thinking the strongest on "this should be in readme". If not, obvious one is the grant to link here (with the permalink, because there might be renames/reorgs and I'd hate for someone having to hunt down broken links): https://github.com/ipfs/devgrants/blob/52232fd6516a6d08db698eee62ea86f7dba5f7ad/open-grants/ipfs-rust/README.md